### ISSUE TITLE
HIP-clang doesn't fully support `-Os`, `-Oz`, and `-Og`

### ISSUE THREAD

**Issue by robertmaynard** [ROCm-Developer-Tools/HIP#2188]
The [ `-Os`, `-Oz`, and `-Og` code generation flags](https://clang.llvm.org/docs/CommandGuide/clang.html#code-generation-options) all produce the following error:

```
"/opt/rocm/llvm/bin/lld" -flavor gnu --no-undefined -shared -plugin-opt=-amdgpu-internalize-symbols -plugin-opt=mcpu=gfx900 -plugin-opt=Og -plugin-opt=-amdgpu-early-inline-all=true -plugin-opt=-amdgpu-function-calls=false -o /tmp/BoundsGlobalCompute-c5b7ea.out /tmp/BoundsGlobalCompute-51cc7e.o
lld: error: -plugin-opt=Og: number expected, but got 'g'
clang-11: error: amdgcn-link command failed with exit code 1 (use -v to see invocation)
```

A mapping between these optimization levels and the appropriate numeric value would resolve this issue ( s => 2, z => 2, g => 1 ). Some form of mapping seems to already exist as `-O` is properly handled

**Comment by yxsamliu** [comment]
which rocm release are you using? The issue seems to have been fixed in rocm 3.9. I saw -Og translated to -plugin-opt=O1

**Comment by robertmaynard** [comment]
I was testing with 3.8. I just re-tested with 3.9 and can confirm that all 3 are now properly supported. Thank you
